buildkite/bazelci.py

```diff
@@ -2130,6 +2148,112 @@ def expand_test_target_patterns(bazel_binary, platform, test_targets):
     return output.split("\n") if output else []
 
 
+def filter_unchanged_targets(
+    expanded_test_targets, workspace_dir, bazel_binary, diffbase, git_commit
+):
+    print_collapsed_group(
+        f":scissors: Filtering targets that haven't been affected since {diffbase}"
+    )
+
+    tmpdir = tempfile.mkdtemp()
+    eprint(f"Downloading bazel-diff to {tmpdir}")
+    bazel_diff_path = download_bazel_diff(tmpdir)
+    resolved_diffbase = resolve_diffbase(diffbase)
+
+    eprint(f"Running bazel-diff for {resolved_diffbase} and {git_commit}")
+    try:
+        affected_targets = run_bazel_diff(
+            bazel_diff_path, workspace_dir, bazel_binary, resolved_diffbase, git_commit, tmpdir
+        )
+    finally:
+        shutil.rmtree(tmpdir)
+
+    filtered_targets = list(set(expanded_test_targets).intersection(affected_targets))
+    if len(filtered_targets) < len(expanded_test_targets):
+        execute_command(
+            [
+                "buildkite-agent",
+                "annotate",
+                "--style=info",
+                "'This run only contains test targets that have been changed since "
+                "{} due to the {} env variable'".format(resolved_diffbase, USE_BAZEL_DIFF_ENV_VAR),
+            ]
+        )
+
+    return filtered_targets
```

Comment:
Is it possible this is an empty set? Is it going to be a problem?

Reply:
Can be empty -> CI will only print the annotation and not run "bazel test"